projects/wave-core-new/src/lib/datasets/dataset.service.ts

```diff
@@ -5,13 +5,14 @@ import {DataSet} from './dataset.model';
 import {UserService} from '../users/user.service';
 import {map, mergeMap} from 'rxjs/operators';
 import {HttpEvent} from '@angular/common/http';
-import {CreateDataSetDict, DataSetIdDict, UploadResponseDict} from '../backend/backend.model';
+import {AutoCreateDataSetDict, CreateDataSetDict, DataSetIdDict, DatasetIdResponseDict, UploadResponseDict} from '../backend/backend.model';
+import {RandomColorService} from '../util/services/random-color.service';
 
 @Injectable({
     providedIn: 'root',
 })
 export class DataSetService {
-    constructor(private backend: BackendService, private userService: UserService) {}
+    constructor(private backend: BackendService, private userService: UserService, private randomColorService: RandomColorService) {}
```

Comment:
Why do you add it and not use it?
If you touch it, you could set the others to protected.

Reply:
I first added the addToMap method there, if it stays in ProjectService (cf other comment) I will remove it.

Reply:
6b05d3ccd0295b16ae979a4cca1f1a468a5d2489